crates/moonbuild/src/build_script.rs

```diff
@@ -0,0 +1,305 @@
+    for link_cfg in output.link_configs {
+        // FIXME: We don't check whether the package and outputs match yet. This
+        // means a module might be able to modify some other package's link config.
+        // This is a bug that needs to be address further down the polish.
+        let Some(pkg) = mdb.get_package_by_name_mut_safe(&link_cfg.package) else {
+            continue;
+        };
+        pkg.link_flags = link_cfg.link_flags;
+        pkg.link_libs = link_cfg.link_libs;
+        pkg.link_search_paths = link_cfg.link_search_paths;
+    }
+}
+
+fn run_replace_in_package(
+    pkg: &mut moonutil::package::Package,
+    env_vars: &HashMap<String, String>,
+    regex: &Regex,
+) -> anyhow::Result<()> {
+    if let Some(link) = pkg.link.as_mut() {
+        if let Some(native) = link.native.as_mut() {
+            if let Some(cc) = native.cc.as_mut() {
+                string_match_and_replace(cc, env_vars, regex).context("when replacing cc")?;
+            }
+            if let Some(cc_flags) = native.cc_flags.as_mut() {
+                string_match_and_replace(cc_flags, env_vars, regex)
+                    .context("when replacing cc_flags")?;
+            }
+            if let Some(cc_link_flags) = native.cc_link_flags.as_mut() {
+                string_match_and_replace(cc_link_flags, env_vars, regex)
+                    .context("when replacing cc_link_flags")?;
+            }
+            if let Some(stub_cc) = native.stub_cc.as_mut() {
+                string_match_and_replace(stub_cc, env_vars, regex)
+                    .context("when replacing stub_cc")?;
+            }
+            if let Some(stub_cc_flags) = native.stub_cc_flags.as_mut() {
+                string_match_and_replace(stub_cc_flags, env_vars, regex)
+                    .context("when replacing stub_cc_flags")?;
+            }
+            if let Some(stub_cc_link_flags) = native.stub_cc_link_flags.as_mut() {
+                string_match_and_replace(stub_cc_link_flags, env_vars, regex)
+                    .context("when replacing stub_cc_link_flags")?;
+            }
+        }
+    }
+    Ok(())
+}
+
+fn string_match_and_replace(
+    s: &mut String,
+    env_vars: &HashMap<String, String>,
+    regex: &Regex,
+) -> anyhow::Result<()> {
+    let mut err = None;
+    let out = regex.replace_all(s, |cap: &Captures| {
+        let name = cap.get(1).expect("failed to get capture group");
+        let name = name.as_str();
+        let Some(value) = env_vars.get(name) else {
+            err = Some(anyhow::anyhow!("Unable to find env var `{}`", name));
+            return "";
+        };
+        value
+    });
+    match out {
+        std::borrow::Cow::Borrowed(_) => {
+            // s is not changed
+        }
+        std::borrow::Cow::Owned(new_s) => {
+            *s = new_s;
+        }
+    }
+    Ok(())
+}
+
+static NODE_EXECUTABLE: LazyLock<Option<&str>> = LazyLock::new(|| {
+    if which::which("node").is_ok() {
```

Comment:
for our own ide in windows, we need `node.cmd`

cc @bzy-debug

https://github.com/moonbitlang/moon/blob/b27a09861027b74d64453eb359cc2bd916ce7f4f/crates/moonbuild/src/build.rs#L46-L50